### ISSUE TITLE
one relation appears many times in beforeChange hook

### ISSUE THREAD

**Issue by Olya-Yer** [keystonejs/keystone-5#1353]
in before change hook one object is represented several times 

in resolved Data 

![image](https://user-images.githubusercontent.com/33322677/60573777-690c3980-9d89-11e9-81d2-acd0c3c1a97d.png)

**Comment by jesstelford** [comment]
Are you able to share your schema & the hook code so we can reproduce the issue?

**Comment by Olya-Yer** [comment]
keystone.createList('Object', {
  schemaDoc: 'A list of things which need to be done',
  fields: {
    // This list is setup with a single field, 'name'.
    // The type of Text influences the types and queries available via GraphQL.
    // See the Text field docs:
    // https://v5.keystonejs.com/keystone-alpha/fields/src/types/text/
    name: { type: Text, schemaDoc: 'This is the thing you need to do', isUnique: true },
    field: { type: Relationship, ref: 'Field', many: true}
  },
  hooks: {
    beforeChange: async ({
      resolvedData,
      existingItem,
      originalInput,
      context,
      list,
    }) => {
      beforeChange({
        resolvedData,
        existingItem,
        originalInput,
        context,
        list,
      })
    }
  }
});
const beforeChange = ({
  resolvedData,
  existingItem,
  originalInput,
  context,
  list,
}) => {
  console.log('resolvedData',resolvedData)
  console.log('existingItem',existingItem)
  console.log('originalInput',originalInput)
  // console.log('context',context)
  console.log('list',list)
};

**Comment by Olya-Yer** [comment]
keystone.createList('Field', {
  schemaDoc: 'A list of things which need to be done',
  fields: {
    name: { type: Text, schemaDoc: 'this is field of ypur object'},
  },
});

keystone.createList('Object', {
  schemaDoc: 'A list of things which need to be done',
  fields: {
    name: { type: Text, schemaDoc: 'This is the object/document/table', isUnique: true },
    field: { type: Relationship, ref: 'Field', many: true}
  },
  hooks: {
    beforeChange: async ({
      resolvedData,
      existingItem,
      originalInput,
      context,
      list,
    }) => {
      beforeChange({
        resolvedData,
        existingItem,
        originalInput,
        context,
        list,
      })
    }
  }
});
const beforeChange = ({
  resolvedData,
  existingItem,
  originalInput,
  context,
  list,
}) => {
  console.log('resolvedData',resolvedData)
  console.log('existingItem',existingItem)
  console.log('originalInput',originalInput)
  // console.log('context',context)
  console.log('list',list)
};

**Comment by Olya-Yer** [comment]
keystone.createList('Field', {
  schemaDoc: 'field of the table',
  fields: {
    name: { type: Text, schemaDoc: 'this is field of ypur object'},
  },
});

keystone.createList('Object', {
  schemaDoc: 'the table structure',
  fields: {
    name: { type: Text, schemaDoc: 'This is the object/document/table', isUnique: true },
    field: { type: Relationship, ref: 'Field', many: true}
  },
  hooks: {
    beforeChange: async ({
      resolvedData,
      existingItem,
      originalInput,
      context,
      list,
    }) => {
      beforeChange({
        resolvedData,
        existingItem,
        originalInput,
        context,
        list,
      })
    }
  }
});
const beforeChange = ({
  resolvedData,
  existingItem,
  originalInput,
  context,
  list,
}) => {
  console.log('resolvedData',resolvedData)
  console.log('existingItem',existingItem)
  console.log('originalInput',originalInput)
  // console.log('context',context)
  console.log('list',list)
};

**Comment by Olya-Yer** [comment]
the bug appears when you add new field one by one and save after each field is added

**Comment by jesstelford** [comment]
This looks like an issue with the way nested creates are being handled in the Admin UI.

A complete reproduction:

```javascript
const { Keystone } = require('@keystone-alpha/keystone');
const { MongooseAdapter } = require('@keystone-alpha/adapter-mongoose');
const { Text, Relationship } = require('@keystone-alpha/fields');
const { GraphQLApp } = require('@keystone-alpha/app-graphql');
const { AdminUIApp } = require('@keystone-alpha/app-admin-ui');

const keystone = new Keystone({
  name: 'test-relationship-app',
  adapter: new MongooseAdapter(),
});

keystone.createList('Field', {
  fields: {
    name: { type: Text },
  },
});

keystone.createList('Object', {
  fields: {
    name: { type: Text, isUnique: true },
    field: { type: Relationship, ref: 'Field', many: true}
  },
  hooks: {
    beforeChange: ({ resolvedData, existingItem }) => {
      console.log('resolvedData',resolvedData)
      console.log('existingItem',existingItem)
    }
  }
});

module.exports = {
  keystone,
  apps: [
    new GraphQLApp(),
    new AdminUIApp({ enableDefaultRoute: true }),
  ],
};
```

## Steps

1. Visit http://localhost:3000/admin
2. Create a `Field` item, 'Bar'
3. Create a `Object` item, linked to the 'Bar' Field.
4. Note the output:
	```
	resolvedData { field: [ '5d1e83daaaa798258fa79c5c' ] } // the `Bar` Field
	existingItem { field: [], _id: 5d1e83d1aaa798258fa79c5b }
	```
5. Click "+" to do a nested create of a new Field item, hit "Create", then on the Object, hit "Save".
5. Note the output:
	```
	resolvedData { field: [ '5d1e83daaaa798258fa79c5c', '5d1e83daaaa798258fa79c5c', '5d1e83e8aaa798258fa79c5d' ] }
	existingItem { field: [ 5d1e83daaaa798258fa79c5c ], _id: 5d1e83d1aaa798258fa79c5b }
	```
7. The `resolvedData` now has a duplicate entry for the existing `Bar` Field item (id `5d1e83daaaa798258fa79c5c`).
8. Note also that the database is storing the item as a duplicate:
	```
	> db.objects.find({})
	{ "_id" : ObjectId("5d1e83d1aaa798258fa79c5b"), "field" : [ ObjectId("5d1e83daaaa798258fa79c5c"), ObjectId("5d1e83daaaa798258fa79c5c"), ObjectId("5d1e83e8aaa798258fa79c5d") ], "name" : "Foo", "__v" : 0 }
	```

**Comment by jesstelford** [comment]
May be related to #1185

**Comment by Olya-Yer** [comment]
if you try to delete one of the added items after the first one, the duplicates will multiply.

**Comment by Olya-Yer** [comment]
This seems fixed  in latest release: #1366

**Comment by jesstelford** [comment]
Confirmed, this is now fixed. Thanks for the report @Olya-Yer! 🎉
